junit-jupiter-api/src/main/java/org/junit/jupiter/api/DynamicTest.java

```diff
@@ -113,4 +116,16 @@ public Executable getExecutable() {
 		return this.executable;
 	}
 
+	public DynamicTest withTags(String... tags) {
+		DynamicTest result = new DynamicTest(getDisplayName(), executable);
+		result.tags = new HashSet<>(Arrays.asList(tags));
```

Comment:
I think calling `withTags()` multiple times should add the tags instead of overwriting the existing ones.